pages/upgrade-guide/4.0.md

```diff
@@ -139,27 +181,33 @@ jQuery 4.0 switches strategies when it comes to setting CSS properties that requ
 
 Instead, jQuery will now only add `px` to a limited set of properties that are known to require it. This change may affect existing code that relies on the old behavior.
 
+#### Breaking change: Remove opacity CSS hook
+
+`.css( "opacity" )` will now return an empty string for detached elements in standard-compliant browsers and "1" in IE. That behavior is shared by most other CSS properties, but may affect existing code relying on a return value of "1".
+
 ### Data
 
 #### Important fix: `Object.prototype` pollution
 
-jQuery 4.0 includes a fix to ensure event and data keys matching `Object.prototype` properties are supported without overriding the native properties. This change is not expected to affect existing code unless it specifically used native property names as event or data keys, which may have caused unexpected behavior anyway.
+jQuery 4.0 includes a fix to ensure event and data keys matching `Object.prototype` properties are supported without overriding the native properties. This change is not expected to affect existing code.
```

Comment:
I've seen my share of `obj.hasOwnProperty( key )` checks; I expect the other `Object.prototype` properties are used way less often. Admittedly, it was more common in the past. How about:
````suggestion
jQuery 4.0 includes a fix to ensure event and data keys matching `Object.prototype` properties are supported without overriding the native properties. Don't call `hasOwnProperty` on a data object:
```js
if ( elem.data().hasOwnProperty( "key" ) ) { /* ... */ }
```
Instead, use on of the following:
```js
if ( "key" in elem.data() ) { /* ... */ }
if ( Object.hasOwn( elem.data(), "key" ) ) { /* ... */ }
```
````
If that's too long, I'd still consider at least a very short note about `hasOwnProperty`.

Reply:
Or maybe use something similar to what you already have in a different section:
> This may affect code calling `hasOwnProperty` on `jQuery.event.special`, which should be rare.